src/models/typeDefinition.ts

```diff
@@ -102,6 +103,44 @@ export abstract class TypeDefinitionProperty {
             this.rawSchemaFormat = "json";
         }
     }
+
+    protected getTypeNameFromRef($ref: string): string {
+        return $ref && $ref.split("/").pop();
+    }
+
+    protected desctructCombination(contract: SchemaObjectContract) {
```

Comment:
Probably typo in the name: `protected desctructCombination(contract: SchemaObjectContract)`. Also, would be great to declare interface for and specify return type.